src/executor.ts

```diff
@@ -658,33 +674,34 @@ class ScriptExecution {
     if (this.#script.output === undefined) {
       return;
     }
-    const absFiles = await glob(this.#script.output.values, {
-      cwd: this.#script.packageDir,
-      absolute: true,
-      followSymlinks: false,
-      includeDirectories: true,
-      expandDirectories: true,
-      rerootToCwd: true,
-    });
-    if (absFiles.length === 0) {
-      return;
-    }
-    const insidePackagePrefix = this.#script.packageDir + pathlib.sep;
-    for (const absFile of absFiles) {
-      // TODO(aomarks) It would be better to do this in the Analyzer by looking
-      // at the output glob patterns, so that we catch errors earlier and can
-      // provide a more useful message, but we need to be certain that we are
-      // parsing glob patterns correctly (e.g. negations and other syntax make
-      // it slightly tricky to detect).
-      if (!absFile.path.startsWith(insidePackagePrefix)) {
+    let absFiles;
+    try {
+      absFiles = await glob(this.#script.output.values, {
+        cwd: this.#script.packageDir,
+        absolute: true,
+        followSymlinks: false,
+        includeDirectories: true,
+        expandDirectories: true,
+        rerootToCwd: true,
+        throwIfOutsideCwd: true,
+      });
+    } catch (error) {
+      if (error instanceof GlobOutsideCwdError) {
```

Comment:
I think, for tools like this that are often misconfigured and where we want to give a clear error message in that case, throwing errors is probably a bad strategy for communicating known failure modes, because it's not particularly type safe or easy to remember / notice.

WDYT about starting to use Result types instead?

Reply:
Might be right, but I think I'd prefer to think about a refactoring like this some other time.

I decided to use use exceptions for errors consistently because it seemed like it simplified error handling. Exceptions can always happen because of APIs we don't control, so they are in the mix regardless. So we use `AggregateError` to accumulate all of the errors we encounter as we execute the build graph (both known and unknown). And then at the top-level, we can just look at all the errors, and nicely format the known ones. I'm not sure that's so obviously bad.